test/e2e/cronjob_test.go

```diff
@@ -57,6 +60,47 @@ func TestSourceCronJob(t *testing.T) {
 	out, err := test.getResourceFieldsWithJSONPath("cronjobsource", "testcronjobsource2", jpSinkRefNameInSpec)
 	assert.NilError(t, err)
 	assert.Equal(t, out, "testsvc1")
+
+	t.Log("verify event messages from cronjob source")
+	mymsg := "This is a message from cronjob."
+	test.serviceCreateEventDisplay(t, r, "displaysvc")
+	test.cronJobSourceCreate(t, r, "testcronjobsource3", "*/1 * * * *", mymsg, "svc:displaysvc")
+	err = test.verifyDeventDisplayLogs(t, "displaysvc", mymsg)
+	assert.NilError(t, err)
+}
+
+func (test *e2eTest) verifyDeventDisplayLogs(t *testing.T, svcname string, message string) error {
+	var (
+		retries int
+		err     error
+		out     string
+	)
+
+	selectorstr := fmt.Sprintf("serving.knative.dev/service=%s", svcname)
+	for retries < 5 {
+		out, err = kubectl{}.Run("logs", "-l", selectorstr, "-c", "user-container")
```

Comment:
I'd separate the retry logic and code into it's on private function.

Would also simplify this function.